coredocument/coredocument.proto

```diff
@@ -169,7 +167,7 @@ message SaltedRole {
 }
 
 message ReadRule {
-  repeated bytes roles = 2;
+  repeated Role roles = 2;
```

Comment:
This is incorrect. The read rule shouldn't embed the `Role` just the `Role.identifier`